test/extended/authorization/rolebinding_restrictions.go

```diff
@@ -0,0 +1,234 @@
+	oc := exutil.NewCLI("rolebinding-restrictions", exutil.KubeConfigPath())
+	g.Context("", func() {
+		g.Describe("Create a rolebinding when there are no restrictions", func() {
+			g.It(fmt.Sprintf("should succeed"), func() {
+				ns := oc.CreateProject()
+				user := "alice"
+				_, err := oc.AdminAuthorizationClient().Authorization().Roles(ns).Create(generateRole(ns))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindings(ns).Create(generateRolebinding(ns, user, "rb1"))
+				o.Expect(err).NotTo(o.HaveOccurred())
+			})
+		})
+
+		g.Describe("Create a rolebinding when subject is permitted by RBR", func() {
+			g.It(fmt.Sprintf("should succeed"), func() {
+				ns := oc.CreateProject()
+				user := "bob"
+				_, err := oc.AdminAuthorizationClient().Authorization().Roles(ns).Create(generateRole(ns))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindingRestrictions(ns).Create(generateAllowUserRolebindingRestriction(ns, user))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindings(ns).Create(generateRolebinding(ns, user, "rb1"))
+				o.Expect(err).NotTo(o.HaveOccurred())
+			})
+		})
+
+		g.Describe("Create a rolebinding when subject is already bound", func() {
+			g.It(fmt.Sprintf("should succeed"), func() {
+				user := "cindy"
+				ns := oc.CreateProject()
+				_, err := oc.AdminAuthorizationClient().Authorization().Roles(ns).Create(generateRole(ns))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindings(ns).Create(generateRolebinding(ns, user, "rb1"))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindingRestrictions(ns).Create(generateAllowUserRolebindingRestriction(ns, user))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindings(ns).Create(generateRolebinding(ns, user, "rb2"))
+				o.Expect(err).NotTo(o.HaveOccurred())
+			})
+		})
+
+		g.Describe("Create a rolebinding when subject is not already bound and is not permitted by any RBR", func() {
+			g.It(fmt.Sprintf("should fail"), func() {
+				ns := oc.CreateProject()
+				user1 := "dave"
+				user2 := "eve"
+				_, err := oc.AdminAuthorizationClient().Authorization().Roles(ns).Create(generateRole(ns))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindings(ns).Create(generateRolebinding(ns, user1, "rb1"))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindingRestrictions(ns).Create(generateAllowUserRolebindingRestriction(ns, user1))
+				o.Expect(err).NotTo(o.HaveOccurred())
+				_, err = oc.AdminAuthorizationClient().Authorization().RoleBindings(ns).Create(generateRolebinding(ns, user2, "rb2"))
```

Comment:
This will flake in CI

Reply:
it hasn't _yet_? I've run it like 50 times since opening this PR. hm... 
suggestions on how to make it un-flaky?

Reply:
You are racing the watch that is feeding the server side informer meaning you will flake every so often.  `wait.PollImmediate` is probably the best you can do.

Reply:
ok used UntilWithSync (I _think_ I used it correctly)